src/odemis/driver/pmtctrl.py

```diff
@@ -447,6 +470,36 @@ def _sendCommand(self, cmd):
 
             return ans.rstrip()
 
+    def _tryRecover(self):
+        self._recovering = True
+        self.state._set_value(HwError("PMTControl disconnected"), force_write=True)
+        # Retry to open the serial port (in case it was unplugged)
+        # _ser_access should already be acquired, but since it's an RLock it can be acquired
+        # again in the same thread
+        with self._ser_access:
+            while True:
+                try:
+                    self._serial.close()
+                    self._serial = None
+                except Exception:
+                    pass
+                try:
+                    logging.debug("retrying to open port %s", self._port)
+                    self._port = self._findDevice(self._portname)
+                except IOError:
+                    time.sleep(2)
+                except Exception:
+                    logging.exception("Unexpected error while trying to recover device")
+                    raise
+                else:
+                    # We found it back!
+                    break
+
+        # it now should be accessible again
+        self.state._set_value(model.ST_RUNNING, force_write=True)
+        logging.info("Recovered device on port %s", self._port)
+        self._recovering = False
```

Comment:
Probably this should be done always when quitting this function, even if an exception is raised. => Move this into a try/finally?

Reply:
I think the _recovering flag should be true as long as we try to find the device to avoid infinite recursion (sendCommand calls tryrecover calls sendcommand calls tryrecover...). Only once there are no exceptions anymore, we want to continue as usual (flat reset to false).

Reply:
I'm worried about the exception happening line 496. If it ever gets triggered, recovering will never happen again.

Reply:
Indeed, I didn't see that. I would suggest to also add a `self._recovering = False` to the except.